File: user-srv/src/Pi.User/Pi.User.Application/Services/LegacyUserInfo/TradingAccount.cs
namespace Pi.User.Application.Services.LegacyUserInfo;

public record TradingAccount
{
    public required string TradingAccountNo { get; init; }
    public required string AccountType { get; init; }
    public required string ExchangeMarketId { get; init; }
    public required string AccountTypeCode { get; init; }
    public required string AccountStatus { get; init; }
    public decimal CreditLine { get; init; }
    public DateOnly? CreditLineEffectiveDate { get; init; }
    public DateOnly? CreditLineEndDate { get; init; }
    public string? MarketingId { get; init; }
    public DateOnly? AccountOpeningDate { get; init; }
};

File: onboard-srv-v2/internal/client/dto/trading_account.go
package client

type TradingAccountsMarketingInfo struct {
	Id               string `json:"id"`
	TradingAccountNo string `json:"tradingAccountNo"`
	AccountType      string `json:"accountType"`
	AccountTypeCode  string `json:"accountTypeCode"`
	ExchangeMarketId string `json:"exchangeMarketId"`
	MarketingId      string `json:"marketingId"`
	EndDate          string `json:"endDate"`
}
